src/pubsub.js

```diff
@@ -105,61 +44,46 @@ module.exports = (node, Pubsub, config) => {
       if (pubsub.listenerCount(topic) === 0) {
         pubsub.unsubscribe(topic)
       }
-
-      if (callback) {
-        return nextTick(() => callback())
-      }
-
-      return Promise.resolve()
     },
 
-    publish: promisify((topic, data, callback) => {
+    publish: (topic, data) => {
       if (!node.isStarted() && !pubsub.started) {
-        return nextTick(callback, errCode(new Error(messages.NOT_STARTED_YET), codes.PUBSUB_NOT_STARTED))
+        throw errCode(new Error(messages.NOT_STARTED_YET), codes.PUBSUB_NOT_STARTED)
       }
 
       try {
         data = Buffer.from(data)
       } catch (err) {
-        return nextTick(callback, errCode(new Error('data must be convertible to a Buffer'), 'ERR_DATA_IS_NOT_VALID'))
+        throw errCode(new Error('data must be convertible to a Buffer'), 'ERR_DATA_IS_NOT_VALID')
       }
 
-      pubsub.publish(topic, data, callback)
-    }),
+      return pubsub.publish(topic, data)
+    },
 
-    ls: promisify((callback) => {
+    getTopics: () => {
       if (!node.isStarted() && !pubsub.started) {
-        return nextTick(callback, errCode(new Error(messages.NOT_STARTED_YET), codes.PUBSUB_NOT_STARTED))
+        throw errCode(new Error(messages.NOT_STARTED_YET), codes.PUBSUB_NOT_STARTED)
       }
 
-      const subscriptions = Array.from(pubsub.subscriptions)
-
-      nextTick(() => callback(null, subscriptions))
-    }),
+      return pubsub.getTopics()
+    },
 
-    peers: promisify((topic, callback) => {
+    getPeersSubscribed: (topic) => {
```

Comment:
Would we ever have pubsub peers that aren't subscribed, if so, are those different? Just trying to get a better understanding of what this is expected to return. Might be good to add some jsdocs to the methods missing them.

Reply:
We can have peers connected with pubsub, but that did not subscribe the topic provided.

I will add jsdocs here and more concrete stuff like examples in the docs PR